WordPress/gutenberg
# Home link: Allow usage outside the navigation block

## What?

Allows the home link block outside the navigation block.

Partial for https://github.com/WordPress/gutenberg/issues/49282
Closes https://github.com/WordPress/gutenberg/issues/35926

## Why?

It is a more convenient way for end-users  to add links to the home page without needing to use a navigation block, site logo, or block bindings API.

## How?

The PR:
- Removes the parent block limitation from block.json.
- Adds `showSubmenuIcon` to `usesContext` in block.json, because compared to for example colors, this attribute it always set on the navigation block, so, it can be used to determine if the Home link block is inside the navigation block or not.
- Conditionally prints the block using `<a>` or `<li>`  depending on where it is placed.

## Testing Instructions
Test this in both a template or template part and in a post or page.

Add a navigation block with a home link block inside.
Add a standalone home link block.

Confirm that both copies of the block work correctly in the editors and front:
- All block settings work.
- The correct element is used depending on where the block is placed.
- The correct class names are used (the stand alone block should not have the navigation item class).
- Confirm that the `aria-current="page"` is still present on the block when it is placed on the home page.

### Testing Instructions for Keyboard


## Feedback needed:
What is the correct element to use when the block is not inside the navigation block?

## Screenshots or screencast
